importexport/musicxml/importmxmlpass2.h

```diff
@@ -130,20 +130,30 @@ class MusicXMLParserLyric {
       };
 
 //---------------------------------------------------------
-//   Notation
+//   Generic Annotation
+//      Most with text base attributes (font-*, placement)
 //---------------------------------------------------------
 
-class Notation {
+class GenericAnnotation {
 public:
-      Notation(const QString& name) { _name = name; }
+      GenericAnnotation(const QString& name, const QString& parent = "",
+                        const SymId& symId = SymId::noSym) { _name = name; _parent = parent; _symId = symId; }
       void addAttribute(const QStringRef name, const QStringRef value);
       QString attribute(const QString& name) const;
       QString name() const { return _name; }
+      QString parent() const { return _parent; }
```

Comment:
What does parent mean for GenericAnnotation?
I can't see where it is used

Reply:
It means the XML parent node of saved annotation element, so it could be "notations", "technical", "articulations", etc. Because we are keeping an array of generic annotations, we need this to make sure that you are referencing the right one later. I didn't use it just because the names were unique enough (but the spec might change in the future).

Reply:
For now, all children of notations have unique names. This has been the case throughout the 16 year history of MusicXML. The upcoming 3.2 release contains no open issues suggesting this may change on short notice. It seems highly unlike to me that this will ever be required. Thus it adds complexity to support something that might happen in future. I'd rather have it removed.

Reply:
my mistake on this one, i actually used this on method `MusicXMLParserNotations::addToScore` in line 6124 and below: https://github.com/musescore/MuseScore/blob/1dd47002c508f9a2740fc09603dde2968fd14b8d/importexport/musicxml/importmxmlpass2.cpp#L6124
to simplify the logic based on its parents instead of checking every element individually.